decidim-templates/spec/system/admin/admin_manages_proposal_answer_templates_spec.rb

```diff
@@ -0,0 +1,307 @@
+      )
+
+      within ".edit_proposal_answer_template" do
+        find("*[type=submit]").click
+      end
+
+      expect(page).to have_admin_callout("problem")
+    end
+  end
+
+  describe "copying a template" do
+    let!(:template) { create(:template, :proposal_answer, organization:) }
+
+    before do
+      visit decidim_admin_templates.proposal_answer_templates_path
+    end
+
+    it "copies the template" do
+      within find("tr", text: translated(template.name)) do
+        click_link "Duplicate"
+      end
+
+      expect(page).to have_admin_callout("successfully")
+      expect(page).to have_content(template.name["en"], count: 2)
+    end
+  end
+
+  describe "destroying a template" do
+    let!(:template) { create(:template, :proposal_answer, organization:) }
+
+    before do
+      visit decidim_admin_templates.proposal_answer_templates_path
+    end
+
+    it "destroys the template" do
+      within find("tr", text: translated(template.name)) do
+        accept_confirm { click_link "Delete" }
+      end
+
+      expect(page).to have_admin_callout("successfully")
+      expect(page).to have_no_i18n_content(template.name)
+    end
+  end
+
+  describe "using a proposal_answer_template" do
+    let(:participatory_process) { create(:participatory_process, title: { en: "A participatory process" }, organization:) }
+    let!(:component) { create(:component, manifest_name: :proposals, name: { en: "A component" }, participatory_space: participatory_process) }
+
+    let(:description) { "Some meaningful answer" }
+    let(:field_values) { { internal_state: "rejected" } }
+    let!(:template) { create(:template, :proposal_answer, description: { en: description }, field_values:, organization:, templatable: component) }
+    let!(:proposal) { create(:proposal, component:) }
+
+    before do
+      visit Decidim::EngineRouter.admin_proxy(component).root_path
+      find("a", class: "action-icon--show-proposal").click
+    end
+
+    it "uses the template" do
+      expect(proposal.reload.internal_state).to eq("not_answered")
+      within ".edit_proposal_answer" do
+        select template.name["en"], from: :proposal_answer_template_chooser
+        expect(page).to have_content(description)
+        click_button "Answer"
+      end
+
+      expect(page).to have_admin_callout("Proposal successfully answered")
+
+      within find("tr", text: proposal.title["en"]) do
+        expect(page).to have_content("Rejected")
+      end
+      expect(proposal.reload.internal_state).to eq("rejected")
+    end
+
+    context "when there are no templates" do
+      before do
+        template.destroy!
+        visit Decidim::EngineRouter.admin_proxy(component).root_path
+        find("a", class: "action-icon--show-proposal").click
+      end
+
+      it "hides the proposal selector" do
+        expect(page).not_to have_select(:proposal_answer_template_chooser)
+      end
+    end
+
+    context "when hides the templates from other components" do
```

Comment:
I'm not so sure about what we're checking in these examples (until the interpolations examples at line 272).  Can you rephrase these descriptions please?

Reply:
changed in 257b85a5a8 . Let's see if now is more clear what we are testing there.

Reply:
much better